documentation/manual/src/main/asciidoc/en-US/modules/redis.asciidoc

```diff
@@ -0,0 +1,1946 @@
+
+[NOTE]
+====
+Hibernate OGM doesn't store null values in Redis,
+setting a value to null will be the same as removing the field
+in the corresponding object in the db.
+====
+
+Hibernate OGM support by default the following types:
+
+* [classname]`java.lang.String`
+
+[source, JSON]
+----
+  { "text" : "Hello world!" }
+----
+
+* [classname]`java.lang.Character` (or char primitive)
+
+[source, JSON]
+----
+  { "delimiter" : "/" }
+----
+
+* [classname]`java.lang.Boolean` (or boolean primitive)
+
+[source, JSON]
+----
+  { "favorite" : true } # default mapping
+  { "favorite" : "T" } # if @Type(type = "true_false") is given
+  { "favorite" : "Y" } # if @Type(type = "yes_no") is given
+  { "favorite" : 1 } # if @Type(type = "numeric_boolean") is given
+----
+
+* [classname]`java.lang.Byte` (or byte primitive)
+
+[source, JSON]
+----
+  { "display_mask" : "70" }
+----
+
+* [classname]`java.lang.Short` (or short primitive)
+
+[source, JSON]
+----
+  { "urlPort" : 80 }
+----
+
+* [classname]`java.lang.Integer` (or int primitive)
+
+[source, JSON]
+----
+  { "stockCount" : 12309 }
+----
+
+* [classname]`java.lang.Long` (or long primitive)
+
+[source, JSON]
+----
+  { "userId" : -6718902786625749549 }
+----
+
+* [classname]`java.lang.Float` (or float primitive)
+
+[source, JSON]
+----
+  { "visitRatio" : 10.4 }
+----
+
+* [classname]`java.lang.Double` (or double primitive)
+
+[source, JSON]
+----
+  { "tax_percentage" : 12.34 }
+----
+
+* [classname]`java.math.BigDecimal`
+
+[source, JSON]
+----
+  { "site_weight" : "21.77" }
+----
+
+* [classname]`java.math.BigInteger`
+
+[source, JSON]
+----
+  { "site_weight" : "444" }
+----
+
+* [classname]`java.util.Calendar`
+
+[source, JSON]
+----
+  { "creation" : "2014-11-18T15:51:26.252Z" }
+----
+
+* [classname]`java.util.Date`
+
+[source, JSON]
+----
+  { "last_update" : "2014-11-18T15:51:26.252Z" }
+----
+
+* [classname]`java.util.UUID`
+
+[source, JSON]
+----
+  { "serialNumber" : "71f5713d-69c4-4b62-ad15-aed8ce8d10e0" }
+----
+
+* [classname]`java.util.URL`
+
+[source, JSON]
+----
+  { "url" : "http://www.hibernate.org/" }
+----
+
+==== Entities
+
+Entities are stored as JSON documents and not as BLOBs
+which means each entity property will be translated into a document field.
+You can use the name property of the [classname]`@Table` and [classname]`@Column` annotations
+to rename the collections and the document's fields if you need to.
+
+Redis has no built-in mechanism for detecting concurrent updates to one and the same document.
+
+
+The following shows an example of an entity and its persistent representation in Redis.
+
+.Example of an entity and its representation in Redis
+====
+[source, JAVA]
+----
+@Entity
+public class News {
+
+    @Id
+    private String id;
+
+    @Version
+    @Column(name="version")
+    private int version;
+
+    private String title;
+
+    private String description;
+
+    //getters, setters ...
+}
+----
+
+[source, JSON]
+----
+{
+    "version": "1",
```

Comment:
1

Will change that. Found also some mismatches in the CouchDB doc, that's the base for the Redis docs. I'll contribute the CouchDB fixes later on.